File: modules/wem-api/src/main/java/com/enonic/wem/api/content/page/region/TextComponentType.java
package com.enonic.wem.api.content.page.region;

public final class TextComponentType
    extends ComponentType
{
    public final static TextComponentType INSTANCE = new TextComponentType();

    private static final TextComponentDataSerializer dataSerializer = new TextComponentDataSerializer();

    private TextComponentType()
    {
        super( "text", TextComponent.class );
    }

    @Override
    public TextComponentDataSerializer getDataSerializer()
    {
        return dataSerializer;
    }
}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/content/page/region/TextComponentType.ts
module api.content.page.region {

    export class TextComponentType extends ComponentType {

        private static INSTANCE: TextComponentType = new TextComponentType();

        constructor() {
            super('text');
        }

        newComponentBuilder(): TextComponentBuilder {
            return new TextComponentBuilder();
        }

        public static get(): TextComponentType {
            return TextComponentType.INSTANCE;
        }
    }

}
